docs/design/interpreter/transformation.md

```diff
@@ -0,0 +1,618 @@
+
+### get_type_from_stack
+* Gets mono type from interp's stack type. uses mono_defaults (src/mono/mono/metadata/class-internals.h) to return types from mono_defaults classes (mapped from stack type)
+* Update to return CoreCLR class handle
+* We might need to extend ICorStaticInfo API to get the default types or implement local helper.
+
+### mono_mint_type
+* Returns interp's mint type for type
+* m_class_is_enumtype -> isEnum
+* mono_class_enum_basetype_internal -> isEnum - underlyingType parameter
+* m_class_get_byval_arg - N.A. on CoreCLR
+
+### interp_create_var_explicit
+* m_class_is_simd_type - check the m_ macros map above
+* mono_class_from_mono_type_internal - N.A., just use CORINFO_CLASS_HANDLE
+
+### interp_create_dummy_var
+* m_class_get_byval_arg (mono_defaults.void_class) -> see get_type_from_stack
+* Update to use CoreCLR void class
+
+### interp_create_ref_handle_var
+* m_class_get_byval_arg (mono_defaults.int_class) -> see get_type_from_stack
+* Update to use CoreCLR int class
+
+### push_var
+* mono_class_from_mono_type_internal - N.A., just use CORINFO_CLASS_HANDLE
+
+### push_mono_type
+* mono_class_from_mono_type_internal - N.A., just use CORINFO_CLASS_HANDLE
+* m_type_is_byref -> asCorInfoType() == CORINFO_TYPE_BYREF
+
+### merge_stack_type_information
+* This uses mono class field in the state, that should be resolved with state containing CoreCLR class handles instead
+
+### handle_branch
+* mono_threads_are_safepoints_enabled -> N.A. on CoreCLR
+* Mono sets this flag by:
+        switch (p) {
+        case MONO_THREADS_SUSPEND_FULL_COOP:
+        case MONO_THREADS_SUSPEND_HYBRID:
+                return TRUE;
+        default:
+                return FALSE;
+        }
+* Always add safepoints (as if the mono_threads_are_safepoints_enabled always returned true)
+
+### two_arg_branch
+* m_class_get_name -> getClassNameFromMetadata
+
+### binary_arith_op
+* m_class_get_name -> getClassNameFromMetadata
+
+### shift_op
+* m_class_get_name -> getClassNameFromMetadata
+
+### get_arg_type_exact
+*  mono_method_signature_internal -> getMethodSig
+*  m_class_get_byval_arg - N.A. on CoreCLR
+
+### load_arg
+* mono_method_signature_internal -> getMethodSig
+  mono_class_from_mono_type_internal - N.A., just use CORINFO_CLASS_HANDLE
+* mono_method_signature_internal (td->method)->pinvoke && !mono_method_signature_internal (td->method)->marshalling_disabled -> (getMethodAttribs(...) & CORINFO_FLG_PINVOKE) && pInvokeMarshalingRequired(...)
+* Simplify code with 2 following calls to just use getClassSize
+* mono_class_native_size
+* mono_class_value_size
+
+### store_arg
+* Same changes as load_arg above
+
+### load_local (TransformData *td, int local)
+* mono_class_from_mono_type_internal - N.A., just use CORINFO_CLASS_HANDLE
+
+### mono_interp_jit_call_supported
+* Return false, we will not support JIT (at least not in first version)
```

Comment:
Note that in some cases 'jit calls' can be calls into AOT-compiled code on mono. It still seems valid to scope this out in the first version though.

Reply:
This is good point, when I was writing this comment, I didn't know yet that that JIT call here means in fact AOT call and that it has JIT in the name for historical reasons.